app/controllers/application_controller.rb

```diff
@@ -73,7 +72,17 @@ def analytics
   end
 
   def analytics_user
-    effective_user || AnonymousUser.new
+    hybrid_user || AnonymousUser.new
+  end
+
+  def hybrid_user
+    return User.find_by(id: session[:doc_capture_user_id]) if !current_user && hybrid_user?
+
+    current_user
+  end
+
+  def hybrid_user?
```

Comment:
Is there any way we can do this without having `hybrid_user?` in application controller? The hybrid flow is pretty small in terms of number of controllers and I'd rather than make the entire application have to know about it.

For example, could we just override `analytics_user` in the hybrid flow controllers?

Reply:
yeah I could move them